swap_meet/vendor.py

```diff
@@ -1,2 +1,88 @@
+from swap_meet.item import Item
+
 class Vendor:
-    pass
\ No newline at end of file
+    '''
+    if passing an inventory into an instance of Vendor,
+    the inventory should be a list of instances of the
+    Item class or its subclasses
+    '''
+    def __init__(self, inventory=None):
+        if inventory is None:
+            inventory = []
+        else:
+            for item in inventory:
+                if not isinstance(item, Item):
+                    raise TypeError
+        self.inventory = inventory
+
+    def add(self, item):
+        if isinstance(item, Item):
+            self.inventory.append(item)
+            return item
+        else:
+            raise TypeError
+    
+    def remove(self, item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        return False
+        
+    def get_by_id(self, item_id):
+        for item in self.inventory:
+            if item.id == item_id:
+                return item
+        return None
+    
+    def swap_items(self, other_vendor, my_item, their_item):
+        if my_item in self.inventory and their_item in other_vendor.inventory:
+            other_vendor.inventory.remove(their_item)
+            self.inventory.append(their_item)
+            self.inventory.remove(my_item)
+            other_vendor.inventory.append(my_item)
```

Comment:
This is a good opportunity to use some of the instance methods you've defined. Consider how your implementation is similar to the lines in the `add` and `remove` methods, and how we might refactor:

```python
            other_vendor.remove(their_item)
            self.add(their_item)
            self.remove(my_item)
            other_vendor.add(my_item)
```

This code passes the tests!